common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -371,6 +371,7 @@ private void record0(Object hint, int recordsToSkip) {
                     int size = lastRecords.size();
                     if (size == 0 || !lastRecords.getLast().equals(value)) {
                         lastRecords.add(value);
+                        size += 1;
```

Comment:
please use the increment operator `++size` if incrementing by 1. Also we should remove from `lastRecords` before adding to avoid re-sizing larger than the max size (which we will never fill).